plugins/woocommerce-admin/client/core-profiler/index.tsx

```diff
@@ -1147,15 +1151,69 @@ export const coreProfilerStateMachineDefinition = createMachine( {
 								completed: true,
 							} );
 						},
-						onDone: {
-							actions: 'redirectToWooHome',
-						},
+						onDone: [
+							{
+								target: 'isJetpackConnected',
+								cond: ( _context ) => {
+									return (
+										_context.pluginsSelected.find(
+											( plugin ) => plugin === 'jetpack'
+										) !== undefined ||
+										_context.pluginsAvailable.find(
+											( plugin: Extension ) =>
+												plugin.key === 'jetpack' &&
+												plugin.is_activated
+										) !== undefined
+									);
+								},
+							},
+							{ actions: 'redirectToWooHome' },
+						],
 					},
 					meta: {
 						component: Loader,
 						progress: 100,
 					},
 				},
+				isJetpackConnected: {
```

Comment:
During the design call it was noticed that there's an unexpected spinner when redirecting to the Jetpack login pages:

https://github.com/woocommerce/woocommerce/assets/27843274/41362a7e-348e-488f-aaa6-1718f6b6c68f

I believe this is because there's no meta.component for the isJetpackConnected and sendToJetpackAuthPage states, so adding them there should fix this issue

Reply:
Good catch! Updated in https://github.com/woocommerce/woocommerce/pull/38674/commits/bced7e54accba2bda036ed6ce73f4be02b747991